source/common/router/rds_impl.h

```diff
@@ -111,6 +115,9 @@ class RdsRouteConfigSubscription : Envoy::Config::SubscriptionCallbacks,
     return route_config_providers_;
   }
   RouteConfigUpdatePtr& routeConfigUpdate() { return config_update_info_; }
+  void maybeCreateInitManager(const std::string& version_info,
```

Comment:
Let's not leak this method into the public API of this class. Is it not possible to test its behavior in the context of onConfigUpdate?

Reply:
> Let's not leak this method into the public API of this class
This is a concrete implementation class, not a public interface. I very strongly prefer testability over encapsulation in such a case.

> Is it not possible to test its behavior in the context of onConfigUpdate?

I'm trying to test this in isolation; the setup is already difficult (and fragile) enough without dragging in another method.

Reply:
Perhaps make it private and make the test class a friend so that it can access it?

Reply:
That would result in a dependency to test code in production code, would it not?

Reply:
I think the friend class statement constitutes a forward reference and needn't ever be declared, but I could be wrong.